src/Stratis.Features.SQLiteWalletRepository/Tables/HDTransactionData.cs

```diff
@@ -174,6 +174,17 @@ FROM   HDTransactionData
             return (balanceData.TotalBalance, balanceData.ConfirmedBalance, balanceData.SpendableBalance);
         }
 
+        // Retrieves a transaction by it's id.
```

Comment:
Are we expecting multiple UTXOs to potentially be returned here? Bearing in mind a `TransactionData` is somewhat badly named - it isn't the complete transaction.

Just in the broader sense, a SQL engine can optimise slightly better if it knows it is definitely only expecting a single result, rather than several. That may not be applicable here.

Reply:
The comment and method name might also be misleading, bearing in mind what I said about UTXOs.

Reply:
This new method, as written, would be accessing the relevant outputs of some transaction that spends to this wallet - somewhat similar to `GetTransactionOutputs`. The latter method is probably no longer usable in its current form due to removal of time from transactions.

Reply:
@quantumagi I've added in the wallet id check.

Reply:
@zeptin I've also renamed the method.